sydent/http/httpclient.py

```diff
@@ -17,25 +17,29 @@
 import json
 import logging
 
-from zope.interface import implementer
 from StringIO import StringIO
-from twisted.internet import defer, reactor, ssl
-from twisted.internet.interfaces import IOpenSSLClientConnectionCreator
-from twisted.internet.abstract import isIPAddress, isIPv6Address
-from twisted.internet.endpoints import HostnameEndpoint, wrapClientTLS
-from twisted.internet._sslverify import _defaultCurveName, ClientTLSOptions
+from twisted.internet import defer, reactor
 from twisted.web.client import FileBodyProducer, Agent, readBody
 from twisted.web.http_headers import Headers
-import twisted.names.client
-from twisted.names.error import DNSNameError
-from OpenSSL import SSL, crypto
-from matrixfederationagent import MatrixFederationAgent
+from sydent.http.matrixfederationagent import MatrixFederationAgent
+
+from sydent.utils.tls import ClientTLSOptionsFactory
```

Comment:
I think you've forgotten to check this in.

(I'd also call it `sydent.federation_tls_options` or something. Calling things `utils` is a sign of failure :). )